wurst/systems/trade/TradeShipSupervisor.wurst

```diff
@@ -44,20 +45,28 @@ public class TradeShipSupervisor
         nextStop = firstStop.getNext()
         ships.put(ship, this)
         moveToNextStop()
-        EventListener.add(ship, EVENT_UNIT_ISSUED_TARGET_ORDER, -> onIssuedOrder())
-        EventListener.add(ship, EVENT_UNIT_ISSUED_POINT_ORDER, -> onIssuedOrder())
-        EventListener.add(ship, EVENT_UNIT_ISSUED_ORDER, -> onIssuedOrder())
+        EventListener.add(ship, EVENT_PLAYER_UNIT_ISSUED_TARGET_ORDER, -> onIssuedOrder())
+        EventListener.add(ship, EVENT_PLAYER_UNIT_ISSUED_POINT_ORDER, -> onIssuedOrder())
+        EventListener.add(ship, EVENT_PLAYER_UNIT_ISSUED_ORDER, -> onIssuedOrder())
 
     function onIssuedOrder()
+        if not this.orderListenerEnabled
+            return
+        if GetIssuedOrderId() != Orders.smart and GetIssuedOrderId() != Orders.move
```

Comment:
What is Orders.smart used for?

Reply:
its the order given on a right click, having multiple actions depending on target